patroni/ha.py

```diff
@@ -78,12 +78,16 @@ def touch_member(self):
         self.dcs.touch_member(json.dumps(data, separators=(',', ':')))
 
     def clone(self, clone_member=None, msg='(without leader)'):
-        if self.state_handler.clone(clone_member):
-            logger.info('bootstrapped %s', msg)
-            cluster = self.dcs.get_cluster()
-            node_to_follow = self._get_node_to_follow(cluster)
-            self.state_handler.follow(node_to_follow, cluster.leader, True)
-        else:
+        try:
+            if self.state_handler.clone(clone_member):
+                logger.info('bootstrapped %s', msg)
+                cluster = self.dcs.get_cluster()
+                node_to_follow = self._get_node_to_follow(cluster)
+                self.state_handler.follow(node_to_follow, cluster.leader, True)
+            else:
+                logger.error('failed to bootstrap %s', msg)
+                self.state_handler.remove_data_directory()
+        except:
```

Comment:
Once again, `clone` method is not supposed to raise any exceptions, therefore there is no need to catch them.